### ISSUE TITLE
[FEATURE] Individual commands for on-the-fly actions rather than choices via plugin options

### ISSUE THREAD

**Issue by redactedscribe** [StrangeGirlMurph/obsidian-wikipedia-search#18]
# Description of the Problem

I think it would be useful if the ability to create an article (the recently added feature) versus inserting the text into the current document was not something you had to configure via the plugin's options, but rather by choosing a different command via the Command Palette. This would be better because then we could choose what we want to happen on-the-fly.

# Description of the Solution

Individual commands for all actions:

- Wikipedia Search: Link Article As Inserted Text
- Wikipedia Search: Link Article As New Note
- Wikipedia Search: Open Article In Obsidian
- Wikipedia Search: Open Article In Browser

These commands could also be bound to individual hotkeys if desired. This is probably better than keeping only two commands because that would require the each to provide the choice to choose between inserting as text vs. saving to a new note, or opening in Obsidian vs. a browser, each time one or the other command is run. The only downside of four commands that if you use all of them frequently, and you want them bound to a hotkey, you need four different hotkeys versus two.

If these four commands were implemented, there would no longer need to be a "creates note" toggle in the plugin's templates section, or the "Open article in browser" option.

Thanks for you consideration.

**Comment by StrangeGirlMurph** [comment]
Well you can choose what you want to happen on-the-fly by creating templates with both options and then selecting the option you want. The thing is most "inline" templates don't really work well for creating new notes. That's why I don't think getting rid of the "creates note" toggle to mark those that do work well with notes is a good idea. I mean I could split things up into two separate template categories. I just don't think many people will use the new note feature which would mean it would be more complicated for the majority. I do agree that more specific commands would probably be a good idea. Idk I am just unsure how many people actually would want to pick between the browser and obsidian on a case by case basis. I just don't have the data. What would be the benefit of opening some articles in the browser and some in Obsidian? I don't really see it.

I have to think about it. Please specify if you think you can make your point clearer or more convincing. Either way: Thank you for your issue! I'll need some time to come up with the best solution :)

**Comment by StrangeGirlMurph** [comment]
I mean I could imagine doing it somewhat like this:

-  Wikipedia Search: Link Article 
(stays as is and gives you all template options)
-  Wikipedia Search: Link Article As Inserted Text 
(only shows template options that are meant for inline inserting - using the toggle)
-  Wikipedia Search: Link Article As New Note
(only shows template options that are meant for a new note - using the toggle)
-  Wikipedia Search: Open Article
(stays as is and goes with the default setting)
-  Wikipedia Search: Open Article In Obsidian
(opens articles in obsidian if surfing is enabled)
-  Wikipedia Search: Open Article In Browser
(opens articles in the browser)

Is that what you had in mind?

**Comment by redactedscribe** [comment]
Does this mean if using "Wikipedia Search: Link Article As Inserted Text" and the toggle is enabled that it will create a note instead of inserting the text? I'm probably not familiar enough with the plugin but it seems like the toggle shouldn't affect the "As Inserted Text" or "As New Note" variants, only the base "Wikipedia Search: Link Article".

**Comment by StrangeGirlMurph** [comment]
No. It's like I wrote. The new commands would basically be the default one with a filter that removes all templates to pick from that don't match the type (inline or note).

**Comment by redactedscribe** [comment]
Right, I think I understand. So I presume there will need to be two default templates now in the settings: One for notes and one for inline texts?

**Comment by StrangeGirlMurph** [comment]
Well no. `Link Article As New Note` would just abort when there is no template which is meant to be for a new note. I think making another template the default would be too much. I think most people won't use the note feature anyways. I'll add the commands later

**Comment by StrangeGirlMurph** [comment]
No. I don't understand. The default template can be changed to create notes as well. You can have a default for both by just creating another template and setting the toggle to true.

**Comment by StrangeGirlMurph** [comment]
Btw my plan for now is to wait for more opinions on this from others.

**Comment by StrangeGirlMurph** [comment]
Doesn't seem like others have a problem with how things are rn. I'll close this. For everyone else: If this resonates with you feel free to reopen this issue.
